### PR TITLE
📖 AWSMachinePools docs and flavor

### PR DESCRIPTION
**Pull request by mytunguyen**
**What this PR does / why we need it**:
Documentation for getting started with AWSMachinePools

### CODE REVIEW

File: docs/machinepools.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+spec:
+  minSize: 1
+  maxSize: 10
+  availabilityZones:
+    - us-east-1
+  awsLaunchTemplate:
+    instanceType: "${AWS_CONTROL_PLANE_MACHINE_TYPE}"
+    sshKeyName: "${AWS_SSH_KEY_NAME}"
+  subnets:
+  - ${SUBNET}
```
Comment (randomvariable) [inline docs/machinepools.md:93]:
```suggestion
  - ${AWS_SUBNET}
```

File: templates/cluster-template-machinepool.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+spec:
+  minSize: 1
+  maxSize: 10
+  availabilityZones: 
+    - "${AWS_REGION}"
+  awsLaunchTemplate:
+    instanceType: "${AWS_CONTROL_PLANE_MACHINE_TYPE}"
+    sshKeyName: "${AWS_SSH_KEY_NAME}"
+  subnets:
+  - ${SUBNET}
```
Comment (randomvariable) [inline templates/cluster-template-machinepool.yaml:101]:
```suggestion
  - ${AWS_SUBNET}
```

Comment (randomvariable) [reply templates/cluster-template-machinepool.yaml:101]: Convention is for AWS specific variables to be prefixed

File: docs/machinepools.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+> MachinePool diverges from MachineDeployment in that the MachineDeployment controller uses
+> MachineSets to achieve the aforementioned desired number of machines and to orchestrate updates
+> to the Machines in the managed set, while MachinePool delegates the responsibility of these
+> concerns to an infrastructure provider specific resource such as AWS Auto Scale Groups, GCP
+> Managed Instance Groups, and Azure Virtual Machine Scale Sets.
+
+> MachinePool is optional and doesn't replace the need for MachineSet/Machine since not every
+> infrastructure provider will have an abstraction for managing multiple machines (i.e. bare metal).
+> Users may always opt to choose MachineSet/Machine when they don't see additional value in
+> MachinePool for their use case.
```
Comment (randomvariable) [inline docs/machinepools.md:30]: Could we clean up the language here and make it more directly relevant to the AWS user base? Probably don't need to discuss the history wrt to v1alpha2, and describe more directly the relation of this implementation of MachinePools to AWS ASGs.

**Comment by randomvariable** [comment]
@mytunguyen could you update, and squash commits when you've got a chance?

File: templates/cluster-template-machinepool.yaml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+kind: Cluster
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: AWSCluster
+    name: "${CLUSTER_NAME}"
+  controlPlaneRef:
+    kind: KubeadmControlPlane
```
Comment (richardcase) [inline templates/cluster-template-machinepool.yaml:12]: It would be great to also create a `eks-machinepool` flavor.

Comment (richardcase) [reply templates/cluster-template-machinepool.yaml:12]: I think this can be done in a follow up issue.

**Comment by randomvariable** [comment]
We definitely need the template change before merging @mytunguyen

File: templates/cluster-template-machinepool.yaml (hunk 4 above)
Comment (mytunguyen) [reply templates/cluster-template-machinepool.yaml:12]: sounds good!

File: docs/machinepools.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+Make sure to set up your AWS environment as described [here](https://cluster-api.sigs.k8s.io/user/quick-start.html).
+
+```shell
+export EXP_MACHINE_POOL=true
+clusterctl init --infrastructure aws
+clusterctl config cluster my-cluster --kubernetes-version v1.16.8 --flavor machinepool > my-cluster.yaml
+```
+
+The template used for this [flavor](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors)
+is located [here](../../templates/cluster-template-machinepool.yaml). For example you can add it to `~/.cluster-api/overrides/infrastructure-aws/0.0.0/cluster-template-machinepool.yaml`
```
Comment (richardcase) [inline docs/machinepools.md:31]: This wouldn't be needed if we add it to the release artifacts.....which is probably better. To do this i think the following is needed:

- update the [release-templates make target ](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/Makefile#L440) to copy the file
- update the [release docs](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/docs/book/src/development/releasing.md) to mention the template file

**Review by richardcase — CHANGES_REQUESTED** [review]

File: templates/cluster-template-machinepool.yaml (hunk 4 above)
Comment (richardcase) [reply templates/cluster-template-machinepool.yaml:12]: Created #2029 for the follow up.

File: docs/machinepools.md (hunk 5 above)
Comment (randomvariable) [reply docs/machinepools.md:31]: I'll fix this as as a follow up.

**Pull request merged**